modules/features/account/src/main/java/au/com/shiftyjelly/pocketcasts/account/onboarding/OnboardingLoginPage.kt

```diff
@@ -140,7 +150,12 @@ internal fun OnboardingLoginPage(
             RowButton(
                 text = stringResource(LR.string.onboarding_log_in),
                 enabled = state.enableSubmissionFields,
-                onClick = { viewModel.logIn(onLoginComplete) },
+                onClick = {
+                    viewModel.updateServerErrorMessage(
+                        if (Network.isConnected(context)) null else networkErrorMessage
+                    )
+                    viewModel.logIn(onLoginComplete)
```

Comment:
Checking the network availability here and then having the view model's login method check if the error message is already null or not feels like it has a bit too many moving parts than would be ideal (for example, you have to call the methods in the right order for everything to work). I prefer to drop the `viewModel.updateServerErrorMessage` method and have the `logIn` method check (and handle) the lack of a network connection directly. We could either add a parameter to that method indicating whether the network is available or update the view model to have access to context so we can check it in there with something along these lines:

``` diff
class OnboardingLogInViewModel @Inject constructor(
     private val podcastManager: PodcastManager,
     private val subscriptionManager: SubscriptionManager,
     private val syncManager: SyncManager,
-) : ViewModel(), CoroutineScope {
+    @ApplicationContext appContext: Context,
+) : AndroidViewModel(appContext as Application), CoroutineScope {
 
     override val coroutineContext: CoroutineContext
         get() = Dispatchers.Default

```

What do you think?

Reply:
> One suggestion that I had in mind is instead of using Network.isConnected() and passing the context everywhere, we could just change the Network object into a class and inject it where ever we want to use it. This way we can directly use the constructor injection for the viewmodel and field injection if we require that in any composable file. We could name the class something along the lines of "NetworkManager".
>
> Sounds good?

👍 Sounds great!